docs/design/mp_arktos.md

```diff
@@ -0,0 +1,172 @@
+	```self.store.vpcs_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.vpcs_arktosnet_store[arktosnet.name][vpc.name] = vpc_object```
+
+The Arktos network object is associated with services in similar ways: 
+    ```self.store.services_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.services_arktosnet_store[arktosnet.name][service.name] = service_object```
+
+The subnet object is associated with pods as: 
+    ```self.store.pods_net_store[net.name] = {}```
+    ```self.store.pods_net_store[net.name][pod.name] = pod_object```
+
+In addition, Mizar already provides key/value sets to store relationships between endpoints/pods, subnets/nets and vpcs: 
+
+* ```nets_vpc_store```
+* ```eps_net_store```
+
+## Analysis
+The following describes in detail on the steps involved during creation of each network object such as services and vpcs. 
+
+### The Arktos network object is associated with a vpcID that already exists:
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc01
+  ip: 10.0.0.0
+  prefix: 16
+  subnets:
+  -  net1
+  -  net2
+```
+* List Kubernetes cluster object / Arktos newtork object
+* Find vpc object that is associated with the vpcID
+* Update object stores: map Arktos network with vpc id and stored as a key/value pair. 
+* If subnet objects (which is equivalent to nets in Mizar language) are also listed, Mizar has object store to specify net objects to vpc object association: ```self.nets_vpc_store[net.vpc][net.name] = net_object``` Thus, the net objects can be linked with arktos network object through ```nets_vpc_store``` and ```vpcs_arktosnet_store``` (mentioned in proposal section)
+
+### The Arktos network object is associated with a vpcID that does not exist yet: 
+```yaml
+spec:
+  type: mizar
+  vpcID: vpc123f3
+  ip: 172.0.0.0
+  prefix: 16
+```
+* List Kubernetes cluster object / Arktos newtork object
+* If not able to find vpc object with specified vpcID, trigger VPC creation workflow to create a new VPC with listed CIDR range. 
+* The following steps are the same as previous section. 
+
+### The Arktos network object is associated with multiple services and the default network is used:
+```yaml
+spec:
+  type: mizar
+```
+* List Kubernetes cluster object / Arktos newtork object
+* In case of using services, vpcID should not listed, and default network is selected
+* Waiting on customer to create service using yaml file:  
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  network: my-network
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+* Create ```my-service```
+* Update object store: map Arktos network with services: ```self.store.services_arktosnet_store[my-network][my-service] = service_object```
+
+###	Pod creation 
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+  network: vpc-1
+  nics:
```

Comment:
If the IP is or subnet is specified by user, which component will assign them?

Reply:
First, we should check if the IP address is within subnet's CIDR range. If true, then the ip will be the pod's ip address, and subnet name listed will be the subnet of the pod.

Reply:
I'm sorry - I missed a keyword.  What I wanted to say is, "if the IP or subnet is **not** specified by user, which component will assign them?"

Reply:
If neither IP or subnet is specified, Mizar will pick the first subnet within ```vpc-1``` and assign pod to that subnet.